### PR TITLE
Add flag to inhibit FOP glyph substitution

### PR DESCRIPTION
**Pull request by dadza**
## Description of the new Feature/Bugfix
When FOP is present on the classpath, it will be automatically used to perform glyph substitutions, which are mandatory for complex layout scripts.

But for scripts where glyph substitutions are optional (e.g. Latin ligatures), it makes sense to have an enable/disable flag because glyph substitutions come with drawbacks such as incorrect text when copying from the document.

The PR introduces a new method `Document.setGlyphSubstitutionEnabled` that can be used to disable glyph substitutions when FOP is present.

Related Issue: #297

### CODE REVIEW

File: openpdf/src/main/java/com/lowagie/text/TextRenderingOptions.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	}
+	
+	/**
+	 * Returns a new instance with the specified document language.
+	 * 
+	 * @param documentLanguage the document language
+	 * @return a new instance with the document language
+	 */
+	public TextRenderingOptions withDocumentLanguage(String documentLanguage) {
+		return new TextRenderingOptions(documentLanguage, glyphSubstitutionEnabled);
```
Comment (asturio) [inline openpdf/src/main/java/com/lowagie/text/TextRenderingOptions.java:75]: Not sure, if creating new instances are necessary.

Maybe something like this is better:
this.documentLanguage = documentLanguage;
return this;

File: openpdf/src/main/java/com/lowagie/text/TextRenderingOptions.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	}
+	
+	/**
+	 * Returns a new instance with the specified glyph substitution enabled flag.
+	 * 
+	 * @param glyphSubstitutionEnabled whether glyph substitution is enabled
+	 * @return a new instance with the glyph substitution enabled flag
+	 */
+	public TextRenderingOptions withGlyphSubstitutionsEnabled(boolean glyphSubstitutionEnabled) {
+		return new TextRenderingOptions(documentLanguage, glyphSubstitutionEnabled);
```
Comment (asturio) [inline openpdf/src/main/java/com/lowagie/text/TextRenderingOptions.java:85]: Same here.

File: openpdf/src/main/java/com/lowagie/text/TextRenderingOptions.java (hunk 1 above)
Comment (dadza) [reply openpdf/src/main/java/com/lowagie/text/TextRenderingOptions.java:75]: My intention was to have TextRenderingOptions as an immutable class, so the fields were declared final and this method returned a new object with the modified documentLanguage value.

But if you prefer mutable object I'm OK with changing the method to modify the current object.

File: openpdf/src/main/java/com/lowagie/text/TextRenderingOptions.java (hunk 2 above)
Comment (dadza) [reply openpdf/src/main/java/com/lowagie/text/TextRenderingOptions.java:85]: See above comment.

Comment (dadza) [reply openpdf/src/main/java/com/lowagie/text/TextRenderingOptions.java]: Changed TextRenderingOptions to be a mutable object.

**Pull request merged**